### PR TITLE
feat(enterprise): disable record deletion

### LINKED ISSUE #2730: Add Enterprise Setting to block deletions from the registries
We may have hit our first case of data loss due to someone deleting a record from the registry.  We should definitely allow the enterprise to turn this behavior on and off via a flag in the enterprise settings.

### PR DESCRIPTION
**Pull request by jniles**
The enterprise now has the option to disable the deletion of records
from the registries.  Deletion transaction from the posting journal is
still supported.

Closes #2730.

_Fig 1: The setting in action_.

### CODE REVIEW

**Comment by jniles** [comment]
@mbayopanda, can I get a review?

File: server/controllers/admin/enterprises.js [hunk 1]
```diff
@@ -22,7 +22,8 @@ exports.list = function list(req, res, next) {
     sql = `
       SELECT id, name, abbr, email, po_box, phone,
         BUID(location_id) AS location_id, logo, currency_id,
-        gain_account_id, loss_account_id, enable_price_lock, enable_prepayments
+        gain_account_id, loss_account_id, enable_price_lock, enable_prepayments,
+        enable_delete_records
```
Comment (mbayopanda) [inline server/controllers/admin/enterprises.js:26]: 👍

**Review by mbayopanda — APPROVED** [review]
LGTM 

bors r+

**Pull request merged**